src/values/ObjectValue.js

```diff
@@ -61,6 +61,30 @@ function isWidenedValue(v: void | Value) {
   return false;
 }
 
+function ToStringPartial(realm: Realm, value: AbstractValue): AbstractValue {
```

Comment:
I also added this string coercion which fixes #1676.

It is also required when the key is not an object, which allows me to test the havocing behavior of these.

I'm not sure where to best put this helper. The normal ToStringPartial is expected to return a string value.

Reply:
This should be move to To. I'm not very partial to the xxxPartial naming pattern. Perhaps call this "ToAbstractString".